### DISCUSSION TITLE
Other  runtime for ts api development

### DISCUSSION THREAD

**Discussion by Lenghak** [encoredev/encore#1593] [General]
### Current Implementation
According to a comment by Simon on dev.to:
> Once the request has been fully processed and decoded, it gets handed over to the Node.js event-loop, and then takes the response from the API handler and writes it back to the client.

Based on this, it appears `encore.ts` utilizes Node.js for API handling after Rust completes the validation phase.

### Performance Consideration
Would switching to alternative JavaScript runtimes like Deno or Bun potentially improve performance metrics, given their reported speed advantages over Node.js?

**Comment by marcuskohlberg** [comment]
We plan on supporting both Deno and Bun in the future, but expect the real-world performance benefits to be very small as Encore.ts already outperforms both thanks to its Rust runtime.

**Reply by Lenghak** [reply]
Sounds promising! Thank you for your update.

**Reply by Monar** [reply]
When can we expect support for Deno? In case of Deno I don't think the speed over node is what makes it desirable :D 

Looking forward to Deno support and in the mean time I need to make some compromises :D

**Reply by marcuskohlberg** [reply]
There's experimental support for Bun now.
To enable the Bun experiment, add `"experiments": ["bun-runtime"]` to your `encore.app` file.
